### PR TITLE
majit: opt-in trait-family method dispatch + census follow-ups

### PR DESCRIPTION
**Pull request by youknowone**
## Summary

Accumulated aheui-census (M2–M3) work on the majit translator, on top of #403.

**Receiver-driven trait-family method dispatch (opt-in, #346).** A ≥2-impl trait's `dyn Trait` receivers can be annotated to a base ClassDef that links the impl subclasses; `getattr` on the base resolves the impl `MethodDesc` family via attrfamily merge, so a trait method call decomposes into JIT primitive dispatch instead of an opaque `CallPureI`. Opt-in per consumer via `register_trait_family`; an empty family is a no-op, so existing production paths are unaffected.

- **annotator: register_trait_family method-member seeding** — seeds the base classdef's method members and merges the impl attrfamilies so a subclass-only method surfaces on the base.
- **majit: opt-in receiver-driven trait-family dispatch** — wires the dispatch through the flowspace adapter / pyre call registry.
- **rtyper: implement access_set=None MultipleFrozenPBCRepr** — implements upstream's `access_set=None` frozen-PBC path (rpbc.py:626–632 + 755–772) that pyre had stubbed as `MissingRTypeOperation`. A base classdef accumulates trait-family method attrs that are uncalled in a partial closure (funcdesc callfamily=None → uncallable Function-kind → `getFrozenPBCRepr`), and the multi-desc no-attrs arm now builds a zero-field `Struct('pbc')` repr cached under a singleton `PbcReprKey::AccessNone`. Purely additive: called production methods still route to `FunctionsPBCRepr`.

**Front-end / scripts follow-ups (post-#403).**
- **front: decline width/sign-ambiguous const binop folds** — narrow-wrap Add/Sub/Mul and unsigned-Shr const folds could yield wrong values; residualize instead of misfolding.
- **front: fold const int-array runtime-index read**
- **scripts: add CrateSpec.charon_args pre-`--` flags**
- **scripts: harden llbc_extract fingerprint edge cases**

## Testing

- `cargo test -p majit-translate --lib`: **2950 passed, 0 failed** (incl. both `register_trait_family_*` tests and the reworked `access_set=None` repr tests).
- aheui census `aheui_census_m0` (aheui-rust `1fc5c34`): phaseB failures **4 → 0**; `LinkedList::swap` reaches the code writer; 23 jitcodes emitted. Remaining blockers are all phaseA (NURSERY static, `__array_repeat`, `bigint::from`, numeric-family `compute_at_fixpoint`) and tracked separately.

Refs #346.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: majit/majit-translate/src/translator/rtyper/rpbc.rs [hunk 1]
```diff
@@ -6340,49 +6347,47 @@ pub fn getFrozenPBCRepr(
...
+    // cached under the `pbc_reprs[None]` singleton slot.
     use crate::translator::rtyper::rtyper::PbcReprKey;
-    if let Some(cached) = rtyper
-        .pbc_reprs
-        .borrow()
-        .get(&PbcReprKey::Access(access_key))
-    {
+    let repr_key = match first {
+        Some(access_key) => PbcReprKey::Access(access_key),
+        None => PbcReprKey::AccessNone,
```
[the remark this answers is not in the document]
Comment (youknowone) [reply majit/majit-translate/src/translator/rtyper/rpbc.rs:6381]: Valid catch — fixed in `05b8a8ea3a`. `MultipleFrozenPBCRepr.convert_desc` now requires a `FrozenDesc` only on the `access_set=Some` path (which reads `attrcache`); the `access_set=None` arm accepts any desc, since its fieldmap is empty and nothing is read off the desc — matching upstream `rpbc.py:769`, which imposes no FrozenDesc requirement. Added a regression test that converts a `FunctionDesc` through the None-access repr.

— *commented by Claude*

File: majit/majit-translate/src/lib.rs [hunk 2]
```diff
@@ -1182,6 +1182,53 @@ fn analyze_pipeline_from_module_paths(
...
+        .pipeline
+        .register_trait_families
+        .iter()
+        .filter_map(
+            |trait_qualified| match trait_impl_owners.get(trait_qualified) {
+                Some(owners) => {
+                    let impl_roots: Vec<String> = owners
+                        .iter()
+                        .map(|owner| owner.rsplit("::").next().unwrap_or(owner).to_string())
+                        .collect();
```
[the remark this answers is not in the document]
Comment (youknowone) [reply majit/majit-translate/src/lib.rs:1205]: Valid — addressed in `d9221a9e87`. The family interning + method seeding key impl subclasses by leaf (`canonical_struct_name`), and that leaf-based keying is pervasive and pre-existing (the receiver match `is_registered_trait_family_leaf` and the struct-field registry are all leaf-keyed — see the parity review's §3 on `mir.rs:6601`). Carrying qualified roots through would require qualifying that whole registry end-to-end, which risks the working path and has no current same-leaf consumer.

So this PR adds a defensive guard instead: when two impls of an opted-in trait collapse to the same leaf, the family registration is skipped and logged (leaving the trait's classdef-less / fail-loud disposition) rather than silently mis-seeding — mirroring the `struct_leaf_counts` bail already used on the single-impl path just below. The full qualified-root interning is left as a follow-up for when a real same-leaf consumer appears.

— *commented by Claude*
